lib/charms/nrpe_exporter/v0/nrpe_exporter.py

```diff
@@ -455,11 +457,17 @@ def _generate_data(self, relation) -> Tuple[list, list]:
 
         return nrpe_endpoints, alerts
 
-    def _generate_alert(self, relation, cmd, id, unit) -> dict:
+    def _generate_alert(self, relation, cmd, id, unit, nagios_host_context) -> dict:
         """Generate an on-the-fly Alert rule."""
         unit_label = re.sub(r"^(.*?)[-_](\d+)$", r"\1/\2", id.replace("_", "-"))
         app_label = re.sub(r"^(.*?)[-_]\d+$", r"\1", id.replace("_", "-"))
 
+        # "nagios_host_content" is needed to extract it from the "id" parameter (target-id)
+        # so that we can correctly relabel juju_application and juju_unit.
+        nagios_host_context = nagios_host_context + "-" if nagios_host_context else ""
+        unit_label = unit_label.replace(nagios_host_context, "")
```

Comment:
I realize I'm being pedantic, but this is how things bite as in the future.

Isn't the `nagios_host_context` always at the beginning? If it is, you should reflect that. Otherwise if the unit label is called `ubuntu-ubuntu-is-amazing/0` and the context is `ubuntu`, `unit_label` will end up being `is-amazing/0`.

Reply:
`unit_label.removeprefix(nagios_host_context)` should do the trick

Reply:
Note that `removeprefix` is Python >= 3.9, which implies Jammy and above. Not sure if it's an issue, just mentioning.

Reply:
`nagios_host_context` was added [in this PR](https://github.com/canonical/charm-nrpe/pull/184) to let us extract it from the unit/application name.
You are right, I forgot the last argument `1` in `.replace` to extract only the first occurence